opentelemetry-instrumentation/tests/test_bootstrap.py

```diff
@@ -87,4 +92,28 @@ def test_run_cmd_install(self):
             [call(i) for i in self.installed_libraries],
             any_order=True,
         )
-        self.assertEqual(self.mock_pip_check.call_count, 1)
+        self.mock_pip_check.assert_called_once()
+
+    @patch("sys.argv", ["bootstrap", "-a", "install"])
+    def test_can_override_available_libraries(self):
+        self.pkg_patcher.stop()
```

Comment:
Would it be better to remove the automatic `pkg_patcher.start/stop()` in `setUp/tearDown` and just call `pkg_patcher.start/stop()` when needed?

Reply:
All the tests but these requires the patching

Reply:
Tried this:

```
diff --git a/opentelemetry-instrumentation/tests/test_bootstrap.py b/opentelemetry-instrumentation/tests/test_bootstrap.py
index ab00effe..ce9c154f 100644
--- a/opentelemetry-instrumentation/tests/test_bootstrap.py
+++ b/opentelemetry-instrumentation/tests/test_bootstrap.py
@@ -63,7 +63,6 @@ class TestBootstrap(TestCase):
         super().setUp()
         self.mock_pip_check = self.pip_check_patcher.start()
         self.mock_pip_install = self.pip_install_patcher.start()
-        self.pkg_patcher.start()
 
     def tearDown(self):
         super().tearDown()
@@ -78,6 +77,7 @@ class TestBootstrap(TestCase):
 
     @patch("sys.argv", ["bootstrap", "-a", "requirements"])
     def test_run_cmd_print(self):
+        self.pkg_patcher.start()
         with patch("sys.stdout", new=StringIO()) as fake_out:
             bootstrap.run()
             self.assertEqual(
@@ -87,6 +87,7 @@ class TestBootstrap(TestCase):
 
     @patch("sys.argv", ["bootstrap", "-a", "install"])
     def test_run_cmd_install(self):
+        self.pkg_patcher.start()
         bootstrap.run()
         self.mock_pip_install.assert_has_calls(
             [call(i) for i in self.installed_libraries],
@@ -96,7 +97,6 @@ class TestBootstrap(TestCase):
 
     @patch("sys.argv", ["bootstrap", "-a", "install"])
     def test_can_override_available_libraries(self):
-        self.pkg_patcher.stop()
         bootstrap.run(libraries=[])
         self.mock_pip_install.assert_has_calls(
             [call(i) for i in default_instrumentations],
@@ -106,7 +106,6 @@ class TestBootstrap(TestCase):
 
     @patch("sys.argv", ["bootstrap", "-a", "install"])
     def test_can_override_available_default_instrumentations(self):
-        self.pkg_patcher.stop()
         with patch(
             "opentelemetry.instrumentation.bootstrap._is_installed",
             return_value=True,
```

but I think I prefer having symmetric setUp and tearDown